source/s3_platform_info.c

```diff
@@ -554,9 +554,12 @@ struct aws_byte_cursor aws_s3_get_ec2_instance_type(struct aws_s3_platform_info_
 }
 
 const struct aws_s3_platform_info *aws_s3_get_platform_info_for_current_environment(
-    struct aws_s3_platform_info_loader *loader) {
+    struct aws_s3_platform_info_loader *loader,
+    bool cached_only) {
     /* getting the instance type will set it on the loader the first time if it can */
-    aws_s3_get_ec2_instance_type(loader);
+    if (!cached_only) {
+        aws_s3_get_ec2_instance_type(loader);
+    }
     /* will never be mutated after the above call. */
     return &loader->lock_data.current_env_platform_info;
```

Comment:
hmmmm, we are touching the lock_data. I haven't dig deep into the implementation, but the comment says "will never be mutated after the above call", but, we can now skip `aws_s3_get_ec2_instance_type`, will it be a possible race condition?

Reply:
Thanks, in theory, yes. In practice, this would not have occurred since the flow is that this API is called first to determine target_throughput, and the S3 client is created later in the Python bindings. I have updated the method to use an API with locks instead, since it can have a race condition in theory.